src/class-processor.ts

```diff
@@ -0,0 +1,60 @@
+import { ClassReflector } from './class-reflector';
+import { CallbackValueInspector } from './handlers/callback-value-inspector';
+import { ObjectLiteralValueInspector } from './handlers/object-literal-value-inspector';
+import { EnumValueInspector } from './handlers/enum-value-inspector';
+import { MultiClassValueInspector } from './handlers/multi-class-value-inspector';
+import { SingleClassValueInspector } from './handlers/single-class-value-inspector';
+import { PrimitiveValueInspector } from './handlers/primitive-value-inspector';
+import { ClassLiteral, ClassType } from './types/class.type';
+import { PropertyDto } from './types/property-dto.interface';
+import { ValueInspector } from './types/value-inspector.interface';
+import { IClassProcessor } from './types/iclass-processor.interface';
+
+import FakerStatic = Faker.FakerStatic;
+
+export class ClassProcessor<T> implements IClassProcessor<T> {
+  private static readonly VALUE_INSPECTORS: ClassType<ValueInspector>[] = [
+    EnumValueInspector,
+    MultiClassValueInspector,
+    SingleClassValueInspector,
+    CallbackValueInspector,
+    ObjectLiteralValueInspector,
+    PrimitiveValueInspector,
+  ];
+
+  public static readonly DEFAULT_LOCALE = 'en';
+
+  public constructor(private readonly faker: FakerStatic, private readonly reflector: ClassReflector, locale: string) {
+    this.faker.setLocale(locale);
+  }
+
+  private handlePropertyValue(propertyDto: PropertyDto): T | T[] {
+    for (const inspectorClass of ClassProcessor.VALUE_INSPECTORS) {
+      const inspector = new inspectorClass(this.faker, this) as ValueInspector;
+
+      if (inspector.shouldInspect(propertyDto)) {
```

Comment:
naming:

inspector => handler / producer
shouldInspect => shouldHandle 
deduceValue => handle / process / generateValue / makeValue / produceValue / ?

Reply:
`produceValue` it is